packages/loader/driver-utils/src/adapters/compression/compressionTypes.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export enum SummaryCompressionAlgorithm {
+	None = 1,
+	LZ4 = 2,
+}
+export enum SummaryCompressionProcessor {
+	SummaryBlob = 1,
+}
+
+export interface ICompressionStorageConfig {
+	algorithm: SummaryCompressionAlgorithm;
+	minSizeToCompress: number;
+	processor: SummaryCompressionProcessor;
+}
+
+export const DefaultCompressionStorageConfig = {
+	algorithm: SummaryCompressionAlgorithm.LZ4,
+	minSizeToCompress: 500,
+	processor: SummaryCompressionProcessor.SummaryBlob,
+};
+
+export const defaultIsUseB64OnCompressed = true;
```

Comment:
Hi @milanro - Previously I commented that prefixing should happen prior to base64 encoding.  Digging into this more, it looks like base64 encoding should be handled by the driver.

This is the layering I would expect: base64 encoding/decoding is a transport-level concern that your code shouldn't have to deal with.  We'll need to figure out the SHA mismatch.  I'll see if I can repro and help.

```ts
			case SummaryType.Blob: {
				value =
					typeof summaryObject.content === "string"
						? {
								type: "blob",
								content: summaryObject.content,
								encoding: "utf-8",
						  }
						: {
								type: "blob",
								content: Uint8ArrayToString(summaryObject.content, "base64"),
								encoding: "base64",
						  };
				break;
```

Reply:
Hi @DLehenbauer - I will implement the suggestions and remove Base64 encoding. It will probably be failing at `summaryTreeUploadManager#writeSummaryBlob`  until some fix is performed but I will push it into this PR.